solr/core/src/java/org/apache/solr/cloud/api/collections/SplitShardCmd.java

```diff
@@ -880,12 +880,12 @@ public static void checkDiskSpace(
     req.setResponseParser(new InputStreamResponseParser("prometheus"));
 
     var cloudClient = (CloudHttp2SolrClient) cloudManager.getSolrClient();
-    var http2Client = (Http2SolrClient) cloudClient.getHttpClient();
+    var httpClient = (HttpJettySolrClient) cloudClient.getHttpClient();
 
-    SolrResponse resp =
-        http2Client.requestWithBaseUrl(parentShardLeader.getBaseUrl(), req::process);
+    SolrResponse resp = httpClient.requestWithBaseUrl(parentShardLeader.getBaseUrl(), req::process);
```

Comment:
I wouldn't be surprised if we deprecate the method you are calling.  It's overloaded; the other one is defined on the base client, is used somewhat widely, and returns a NamedList.  Looking at the code here, you only want a NamedList, not SolrResponse specifically.